schedule/schedule.go

```diff
@@ -0,0 +1,165 @@
+package schedule
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+)
+
+const (
+	DateFormat = "Mon 02 Jan 2006"
+)
+
+// Schedule represents a series of Shifts, in temporal order.
+type Schedule struct {
+	// Shifts is the list of shifts in order. The last Shift, and only the last Shift, should have a StopTime value.
+	Shifts []*Shift `json:"shifts"`
+}
+
+func (sch *Schedule) LastShift() *Shift {
+	if sch.Shifts == nil || len(sch.Shifts) == 0 {
+		return nil
+	}
+	return sch.Shifts[len(sch.Shifts)-1]
+}
+
+func (sch *Schedule) Validate() error {
+	if sch == nil {
+		return fmt.Errorf("schedule cannot be nil")
+	}
+
+	if sch.Shifts == nil || len(sch.Shifts) == 0 {
```

Comment:
I'd probably write this as `if sch.LastShift() == nil`, but I admit that's debatable, so up to you...

Reply:
Done. There was actually a bug in my test that was exposed from changing this. Thanks!